mod/metrics/set/abstract/known_answers.rb

```diff
@@ -51,21 +54,35 @@ def researched_relation
   Answer.select(:record_id).distinct.where where_answer
 end
 
-view :research_progress_bar, cache: :never do
-  progress_bar(
-    { value: card.percent_known, class: "progress-known" },
-    { value: card.percent_unknown, class: "progress-unknown" },
-    value: card.percent_not_researched, class: "progress-not-researched"
-  )
+def smart_count
+  return 0 unless worth_counting?
+  yield
 end
 
-view :absolute_research_progress_bar, cache: :never do
-  progress_bar(
-    { value: card.percent_known, label: card.num_known,
-      class: "progress-known" },
-    { value: card.percent_unknown, label: card.num_unknown,
-      class: "progress-unknown" },
-    value: card.percent_not_researched, label: card.num_not_researched,
-    class: "progress-not-researched"
-  )
+format :html do
+  def research_progress_bar link_method=nil
```

Comment:
beyond the api changes, the biggest difference here is that:
1. each section of the progress bar links to answers with the right value filter (eg, a 5 in a "known" section will link to a page filtering for only known answers and should show you those 5)
2. I removed all the zeroes.

As I understand it, this was the original design.